dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/ranger/RangerClient.java

```diff
@@ -105,60 +147,202 @@ public ImmutableList<Object> findPolicies(Map<String, String> args) throws Range
+
+    private final String basicAuth;
+    private final URIBuilder baseUri;
+    private final SSLSocketFactory socketFactory;
+    private final HostnameVerifier hostnameVerifier;
+
+    static class Builder {
+      private String basicAuth;
+      private final URIBuilder baseUri;
+      private SSLSocketFactory socketFactory;
+      private HostnameVerifier hostnameVerifier;
+
+      Builder(URIBuilder baseUri) {
+        this.baseUri = baseUri;
+      }
+
+      Builder withBasicAuth(String basicAuth) {
+        this.basicAuth = basicAuth;
+        return this;
+      }
+
+      ConnectionWrapper build() {
+        return new ConnectionWrapper(baseUri, basicAuth, socketFactory, hostnameVerifier);
+      }
+
+      public Builder disableTlsValidation() {
+        try {
+          TrustManager[] trustAllCerts =
+              new TrustManager[] {
+                new X509TrustManager() {
+                  public java.security.cert.X509Certificate[] getAcceptedIssuers() {
+                    return null;
+                  }
+
+                  public void checkClientTrusted(
+                      java.security.cert.X509Certificate[] certs, String authType) {}
+
+                  public void checkServerTrusted(
+                      java.security.cert.X509Certificate[] certs, String authType) {}
+                }
+              };
+          SSLContext sc = SSLContext.getInstance("SSL"); // Or TLS
+          sc.init(null, trustAllCerts, new java.security.SecureRandom());
+          socketFactory = sc.getSocketFactory(); // Apply globally (affects all connections)
+
+          hostnameVerifier = (hostname, session) -> true;
+        } catch (NoSuchAlgorithmException | KeyManagementException e) {
+          throw new RangerException("Cannot disable tls validation", e);
+        }
+        return this;
+      }
+    }
+
+    public static Builder builder(URIBuilder baseUri) {
+      return new Builder(baseUri);
+    }
+
+    private ConnectionWrapper(
+        URIBuilder baseUri,
+        String basicAuth,
+        SSLSocketFactory socketFactory,
+        HostnameVerifier hostnameVerifier) {
+      this.basicAuth = basicAuth;
+      this.baseUri = baseUri;
+      this.socketFactory = socketFactory;
+      this.hostnameVerifier = hostnameVerifier;
+    }
+
+    public String doGet(String path, Map<String, String> queryParams) {
+      URL url = null;
+      try {
+        url =
+            baseUri
+                .setPath(path)
+                .addParameters(
+                    queryParams.entrySet().stream()
+                        .map(entry -> new BasicNameValuePair(entry.getKey(), entry.getValue()))
+                        .collect(toImmutableList()))
+                .build()
+                .toURL();
+      } catch (Exception ex) {
+        throw new RangerException("Failed to build URL", ex);
       }
-      HttpEntity entity = httpResponse.getEntity();
+      HttpURLConnection connection = null;
```

Comment:
Short version should work as well.
```suggestion
      HttpURLConnection connection;
```

Reply:
It would but with a null check in the finally block I think this is more readable.